include/preprocess/max_inscribed_ellipsoid.hpp

```diff
@@ -84,7 +95,7 @@ std::pair<std::pair<MT, VT>, bool> max_inscribed_ellipsoid(MT A, VT b, VT const&
                 vec_iter3++;
             }
             Q *= (t * t);
-            Y *= (1.0 / (t * t));
+            Y = Y * (1.0 / (t * t));
```

Comment:
Since Y is a diagonal matrix, could this operation be further optimized?

Reply:
same here also, `.asDiagonal()` returns a diagonal matrix for which computations will be optimized.
I read about them from here: [stackoverflow-answer](https://stackoverflow.com/a/45133662)